amulet/deployer.py

```diff
@@ -59,16 +48,31 @@ def __init__(self, juju_env=None, series='precise',
         self.charm_name = get_charm_name(os.getcwd())
 
         self.sentry = None
-        self.deployer = juju_deployer
-        self.deployer_dir = tempfile.mkdtemp(prefix='amulet_deployment_')
+        self.deployer = path(juju_deployer)
+        self.deployer_dir = tempdir(prefix='amulet_deployment_')
 
         if 'JUJU_TEST_CHARM' in os.environ:
             self.charm_name = os.environ['JUJU_TEST_CHARM']
 
         self.charm_cache = CharmCache(self.charm_name)
 
-    def load(self, deploy_cfg):
-        schema = next(iter(deploy_cfg.values()))
+    @classmethod
+    def from_bundle(cls, bundle_file, deployment_name=None):
+        deployment = cls()
+        bundle_file = path(bundle_file)
+        deployment.load_bundle_file(bundle_file, deployment_name)
+        return deployment
+
+    def load_bundle_file(self, bundle_file, deployment_name=None):
+        with open(bundle_file, 'r') as stream:
+            contents = yaml.safe_load(stream)
+        return self.load(contents, deployment_name)
+
+    def load(self, deploy_cfg, deployment_name=None):
```

Comment:
FWIW, deployment_name will be going away with the new bundle format. Thanks for adding this though.